paf-mvp-demo-express/src/websites/portal.ts

```diff
@@ -0,0 +1,297 @@
+    removeCookie(req, res, Cookies.identifiers, { domain: tld });
+    const homeUrl = getRequestUrl(req, '/');
+    httpRedirect(res, homeUrl.toString());
+  });
+
+  portalWebSiteApp.expressApp.get(removePrefsUrl, (req, res) => {
+    removeCookie(req, res, Cookies.preferences, { domain: tld });
+    const homeUrl = getRequestUrl(req, '/');
+    httpRedirect(res, homeUrl.toString());
+  });
+
+  portalWebSiteApp.expressApp.get(generateNewId, (req, res) => {
+    const returnUrl = getRequestUrl(req, writeNewId);
+
+    // First go to "read or init id" on operator, and then redirects to the local write endpoint, that itself calls the operator again
+    httpRedirect(res, client.getReadRedirectUrl(req, returnUrl).toString());
+  });
+
+  portalWebSiteApp.expressApp.get(writeNewId, (req, res) => {
+    const cookies = req.cookies;
+
+    const redirectGetIdsPrefsResponse = getPafDataFromQueryString<RedirectGetIdsPrefsResponse>(req);
+    const identifiers = redirectGetIdsPrefsResponse.response.body.identifiers;
+    const homeUrl = getRequestUrl(req, '/');
+
+    const preferences =
+      // little trick because we know the cookie is available in the same TLD+1
+      typedCookie<Preferences>(cookies[Cookies.preferences]) ??
+      // Assume opt out by default if no preferences
+      client.buildPreferences(identifiers, { use_browsing_for_personalization: false });
+
+    httpRedirect(res, getWritePrefsUrl(req, identifiers, preferences, homeUrl).toString());
+  });
+
+  portalWebSiteApp.expressApp.get(optInUrl, (req, res) => {
+    const cookies = req.cookies;
+    const identifiers = typedCookie<Identifiers>(cookies[Cookies.identifiers]);
+
+    const homeUrl = getRequestUrl(req, '/');
+    if (identifiers) {
+      httpRedirect(res, getWritePrefsUrlFromOptin(req, identifiers, true, homeUrl).toString());
+    } else {
+      // Shouldn't happen: redirect to home page
+      httpRedirect(res, homeUrl.toString());
+    }
+  });
+
+  portalWebSiteApp.expressApp.get(optOutUrl, (req, res) => {
+    const cookies = req.cookies;
+    const identifiers = typedCookie<Identifiers>(cookies[Cookies.identifiers]);
+
+    const homeUrl = getRequestUrl(req, '/');
+    if (identifiers) {
+      httpRedirect(res, getWritePrefsUrlFromOptin(req, identifiers, false, homeUrl).toString());
+    } else {
+      // Shouldn't happen: redirect to home page
+      httpRedirect(res, homeUrl.toString());
+    }
+  });
+
+  const requestWithoutBodyVerifier = (request: RequestWithContext<MessageBase>) =>
+    new RequestVerifier(keyStore.provider, new RequestWithoutBodyDefinition()).verifySignature(request);
+  const postIdsPrefsRequestVerifier = (request: RequestWithContext<PostIdsPrefsRequest>) =>
+    new RequestVerifier(keyStore.provider, new RequestWithBodyDefinition()).verifySignature(request);
+  const responseVerifier = (response: ResponseType) =>
+    new ResponseVerifier(keyStore.provider, new ResponseDefinition()).verifySignature(response);
+  /*
```

Comment:
Commented code. 

If it is meaningful to keep it, maybe a bit of explanation inside it.